From da06c61cdceae1ba02e392d9f18059f087287352 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:32:01 -0400
Subject: [scsi] hpsa: fix event filtering to prevent excessive rescans with old firmware

Message-id: <1410186734-19986-71-git-send-email-thenzl@redhat.com>
Patchwork-id: 92102
O-Subject: [RHEL6.6 e-stor PATCH 70/83] hpsa: fix event filtering to prevent excessive rescans with old firmware
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

CTLR_STATE_CHANGE_EVENT and CTLR_STATE_CHANGE_EVENT_REDUNDANT_CNTRL
do not require rescans to be initiated.  Current firmware filters out
these events already, but some out of date firmware doesn't, so the
driver needs to filter them out too.  Without this change and with out
of date firmware you may see the driver spending a lot of time
scanning devices unnecessarily on some Smart Arrays.

Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Reviewed-by: Justin Lindley <justin.lindley@hp.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit 7b2c46ee7259c52cf2c1f82ab100e9aaeb83f78a)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.h b/drivers/scsi/hpsa.h
index 0a5d91c..24472ce 100644
--- a/drivers/scsi/hpsa.h
+++ b/drivers/scsi/hpsa.h
@@ -229,11 +229,9 @@ struct ctlr_info {
 #define CTLR_STATE_CHANGE_EVENT_AIO_CONFIG_CHANGE (1 << 31)
 
 #define RESCAN_REQUIRED_EVENT_BITS \
-  (CTLR_STATE_CHANGE_EVENT | \
-  CTLR_ENCLOSURE_HOT_PLUG_EVENT | \
+  (CTLR_ENCLOSURE_HOT_PLUG_EVENT | \
   CTLR_STATE_CHANGE_EVENT_PHYSICAL_DRV | \
   CTLR_STATE_CHANGE_EVENT_LOGICAL_DRV | \
-  CTLR_STATE_CHANGE_EVENT_REDUNDANT_CNTRL | \
   CTLR_STATE_CHANGE_EVENT_AIO_ENABLED_DISABLED | \
   CTLR_STATE_CHANGE_EVENT_AIO_CONFIG_CHANGE)
  spinlock_t offline_device_lock;
-- 
1.7.1